lib/queue.rb

```diff
@@ -1,31 +1,57 @@
 class Queue
-
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = []
+    @head_ind = @tail_ind = 0
+    @max_ind = 19 # max size of 20 per README
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    raise ArgumentError, 'Queue is full' if @tail_ind - @head_ind > @max_ind
+
+    @store << element
```

Comment:
⚠️ This won't allow the wrap-around effect of a circular buffer.

Reply:
Ah - I meant to add abs()